Add features to StatusCard

NavyAdmiral opened the pull request:
Old
![image](https://user-images.githubusercontent.com/40662484/59960842-af65bc80-949c-11e9-9c10-b6abf11e4404.png)
![image](https://user-images.githubusercontent.com/40662484/59960950-4bdc8e80-949e-11e9-9e90-e25f6c0fc311.png)

New
![image](https://user-images.githubusercontent.com/40662484/59960914-d8d31800-949d-11e9-8668-96fc5b9031d1.png)
![image](https://user-images.githubusercontent.com/40662484/59960948-41ba9000-949e-11e9-9239-44c8d1c2b455.png)


- Some 💅 polishes
- Displays link earned only for Completed jobs (0 if Amount is not available from RunResult)
- Displays time elapsed for Completed and Errored jobs (does not display for pending jobs)
- ∴ Pending jobs will stay the same as old, as neither link earned nor time elapsed will be available for those

NavyAdmiral commented on operator_ui/src/components/Icons/Success.js:
```diff
@@ -16,7 +16,7 @@ const Success = props => (
     <title>{'Artboard 1'}</title>
     <circle cx={24} cy={24} r={22} fill="none" stroke="#fff" strokeWidth={4} />
-    <circle data-name="path-1" cx={24} cy={24} r={20} fill="#e8faf1" />
+    <circle cx={24} cy={24} r={22} fill="none" stroke={props.isForStatusCard ? '#17c671' : '#fff'} strokeWidth={4} />
```
@rupurt I don't think you're going to love what I've done here and https://github.com/smartcontractkit/chainlink/pull/1392/files#diff-248756b17013f8d03720fdf37721d349R101

rupurt commented on operator_ui/__tests__/components/JobRuns/StatusCard.test.js:
```diff
@@ -1,8 +1,32 @@
+import isoDate from 'test-helpers/isoDate'
 
 describe('components/JobRuns/StatusCard', () => {
+  const minuteAgo = isoDate(Date.now() - 60)
```
I believe this is 60ms. Can you use the `MINUTE_MS` & `TWO_MINUTE_MS` constants from the `isoDate` module

rupurt commented on operator_ui/src/components/Icons/Success.js:
```diff
@@ -15,7 +21,7 @@ const Success = props => (
     </defs>
     <title>{'Artboard 1'}</title>
-    <circle cx={24} cy={24} r={22} fill="none" stroke="#fff" strokeWidth={4} />
+    <circle cx={24} cy={24} r={22} fill="none" stroke={isForStatusCard ? '#17c671' : "#fff"} strokeWidth={4} />
```
I feel like we should pass the color in as a prop and default to `#fff` instead of coupling this to status cards

rupurt commented on operator_ui/src/components/JobRuns/StatusCard.tsx:
```diff
@@ -36,17 +38,56 @@ const styles = (theme: Theme) =>
-      paddingLeft: theme.spacing.unit * 2,
-      textTransform: 'capitalize'
+      textTransform: 'capitalize',
+      color: '#3d5170'
```
This should use theme customization https://material-ui.com/guides/typescript/#customization-of-theme

rupurt commented on operator_ui/src/components/JobRuns/StatusCard.tsx:
```diff
@@ -36,17 +38,56 @@ const styles = (theme: Theme) =>
+      paddingLeft: theme.spacing.unit * 2
+    },
+    elapsedText: {
+      color: '#818ea3'
```
Same as ^

rupurt commented on operator_ui/src/components/JobRuns/StatusCard.tsx:
```diff
@@ -36,17 +38,56 @@ const styles = (theme: Theme) =>
+      color: '#818ea3'
+    },
+    earnedLink: {
+      color: '#17c671'
```
Same as ^

rupurt commented on operator_ui/src/components/JobRuns/StatusCard.tsx:
```diff
@@ -36,17 +38,56 @@ const styles = (theme: Theme) =>
+    <>
+      {jobRun && jobRun.finishedAt && jobRun.createdAt && (
+        <Typography className={classes.elapsedText} variant="h6">
+          {!!hours && `${hours}h`}
```
How do you feel about extracting a util function that can be tested in isolation for this date string formatting and then an `<ElapsedTime>` component? This will also be useful in the explorer.

rupurt commented on operator_ui/src/components/JobRuns/StatusCard.tsx:
```diff
@@ -55,12 +96,24 @@ const StatusCard = ({ title, classes, children }: IProps) => {
+            {renderElapsedTime(jobRun, classes)}
+          </Grid>
+          <Grid item xs={3}>
+            {title === 'completed' && renderLinkEarned(classes, jobRun)}
```
How do you feel about rendering a component rather than a function so that we can ignore parameter order? e.g. `<EarnedLink  ... />`

rupurt commented:
@NavyAdmiral looks like it needs a rebase

rupurt commented on operator_ui/src/components/JobRuns/StatusIcon.js:
```diff
@@ -3,9 +3,15 @@ import SuccessIcon from 'components/Icons/Success'
+      <SuccessIcon
+        width={width}
+        height={height}
+        color={successIconColor}
```
Is there a case where we want a different color for the success icon? It feels like that should be encapsulated within the `<SuccessIcon />` component and should be pulled from the theme, rather than passed in as a prop.

NavyAdmiral replied:
All that was an opinionated visual change of mine anyway, so I've decided to opt out from it. Ptal again @rupurt

rupurt commented:
@NavyAdmiral I think this needs to be rebased on `develop` instead of `master` to get CI passing

NavyAdmiral commented:
@rupurt done

rupurt commented:
@NavyAdmiral can you test this against https://github.com/smartcontractkit/chainlink/tree/chores/operator-ui-strict-typescript and make sure there are no type checking errors.

Currently type checking is not getting performed correctly in the Operator UI so I don't want to merge this unless it's going to pass type checking in that branch.

NOTE: I don't need you to submit the PR with that branch rebased. Just to check that there are no additional type checking errors.

NavyAdmiral commented:
@rupurt There were many unrelated issues and errors popping up when I switched to the strict TS branch. I fixed the errors that I saw through with greps and manual inspection. Lmk your thoughts. Also should I move some of the newly introduced functions to styleguide or should that wait?

NavyAdmiral commented:
Closing in favor of #1457
